neopixel_spi.py

```diff
@@ -95,7 +104,10 @@ def __init__(
         if not pixel_order:
             pixel_order = GRB if bpp == 3 else GRBW
         else:
-            bpp = len(pixel_order)
+            if isinstance(pixel_order, tuple):
```

Comment:
What if it's a string instead of a tuple?

Strings are currently used in this library, but I'm not sure what the current desire is for `pixel_order`? tuple? string? both?

Reply:
I should have left that line outside of the tuple check. That way when `pixel_order` is supplied we always use its length as `bpp`. Then we check if it is a tuple, if yes, it gets converted to a string. And if it is already a string it is good to go as is.

This fixes the problem you were seeing for RBGW as well.